File: Problems/1143.Longest-Common-Subsequence/LongestCommonSubsequence_test.py

from LongestCommonSubsequence import Solution
import pytest
from copy import deepcopy

SOL = Solution()

TEST_SUITS =  [
    ("babcde", "acxe", 3),
    ("babccde", "acxe", 3),
    ("oxcpqrsvwf", "shmtulqrypy", 2),
    ("abcba", "abcbcba", 5),
]

@pytest.mark.parametrize(
    "text1, text2, ans", 
    deepcopy(TEST_SUITS)
)
def test(text1, text2, ans):
    assert SOL.longestCommonSubsequence(text1, text2) == ans
    

File: Problems/1143.Longest-Common-Subsequence/LongestCommonSubsequence_test.go
package main

import "testing"

func Test_longestCommonSubsequence(t *testing.T) {
	type args struct {
		text1 string
		text2 string
	}
	tests := []struct {
		name string
		args args
		want int
	}{
		// TODO: Add test cases.
		{"Ex0", args{"babcde", "acxe"}, 3},
		{"Ex0", args{"babccde", "acxe"}, 3},
		{"Ex0", args{"oxcpqrsvwf", "shmtulqrypy"}, 2},
		{"Ex0", args{"abcba", "abcbcba"}, 5},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := longestCommonSubsequence(tt.args.text1, tt.args.text2); got != tt.want {
				t.Errorf("longestCommonSubsequence() = %v, want %v", got, tt.want)
			}
		})
	}
}
